nix/README.md

```diff
@@ -0,0 +1,77 @@
+# Nix development and build environment
+
+With Nix it's possible to quickly and reliably recreate the full environments
+for developing, testing and building PostgREST.
+
+## Getting started with Nix
+
+You'll need to [get Nix](https://nixos.org/download.html). The installer will
+create your Nix store in the `/nix/` directory, where all build artifacts and
+their dependencies will be stored. It will also link the Nix executables like
+`nix-env`, `nix-build` and `nix-shell` into your PATH. Nix will manage all
+other PostgREST dependencies from here on out. To clean up older build
+artifacts from the `/nix/store`, you can run `nix-collect-garbage`.
+
+## Building PostgREST
+
+To build PostgREST from your local checkout of the repository, run:
+
+```bash
+nix-build --attr postgrest
```

Comment:
Hmm.. somehow this is also building Haddock docs(build take longer to complete). Is that intended?

```
[17 of 17] Compiling PostgREST.App    ( src/PostgREST/App.hs, dist/build/PostgREST/App.p_o )
Preprocessing executable 'postgrest' for postgrest-7.0.0..
Building executable 'postgrest' for postgrest-7.0.0..
[1 of 2] Compiling UnixSocket       ( main/UnixSocket.hs, dist/build/postgrest/postgrest-tmp/UnixSocket.o )
[2 of 2] Compiling Main             ( main/Main.hs, dist/build/postgrest/postgrest-tmp/Main.o )
Linking dist/build/postgrest/postgrest ...
haddockPhase
Preprocessing library for postgrest-7.0.0..
Running Haddock on library for postgrest-7.0.0..
Warning: The documentation for the following packages are not installed. No
links will be generated to these packages: mtl-compat-0.2.2, vault-0.3.1.4
Warning: --source-* options are ignored when --hyperlinked-source is enabled.

Haddock coverage:
Warning: 'limit' is out of scope.
    If you qualify the identifier, haddock can try to link it anyway.
Warning: 'offset' is out of scope.
    If you qualify the identifier, haddock can try to link it anyway.
   9% (  1 / 11) in 'PostgREST.RangeQuery'
  Missing documentation for:
    rangeParse (src/PostgREST/RangeQuery.hs:33)
    rangeRequested (src/PostgREST/RangeQuery.hs:45)
    rangeLimit (src/PostgREST/RangeQuery.hs:54)
    rangeOffset (src/PostgREST/RangeQuery.hs:60)
    restrictRange (src/PostgREST/RangeQuery.hs:48)
    rangeGeq (src/PostgREST/RangeQuery.hs:66)
    allRange (src/PostgREST/RangeQuery.hs:70)
    NonnegRange (src/PostgREST/RangeQuery.hs:31)
    rangeStatusHeader (src/PostgREST/RangeQuery.hs:77)
    contentRangeH (src/PostgREST/RangeQuery.hs:92)
```

Reply:
We could turn that off by adding `dontHaddock` to the `dontCheck` line in [default.nix](https://github.com/PostgREST/postgrest/pull/1489/files#diff-5712e736e0de6ba170577f8472c398e9R35) if we wanted (those helpers are defined in [`haskell-modules/lib.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/haskell-modules/lib.nix)).

But I would leave that on for now, as that's the standard behavior of Haskell builds in Nix.

Reply:
Sorry, still a Nix newbie, how do I add the `dontHaddock`?

```
    pkgs.haskell.lib.dontCheck drv;
```

Reply:
`pkgs.haskell.lib.dontHaddock (pkgs.haskell.lib.dontCheck drv);` should work! 

Or you could also write:

`with pkgs.haskell.lib; dontHaddock (dontCheck drv);` - which is shorter, but it's not so clear anymore from what scope those functions are coming from...

Reply:
Thanks! It's working good now. I'll save that one for myself. I'm fine with keeping the standard behavior.